pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -39,9 +40,9 @@ func ValidateCertificateSpec(crt *cmapi.CertificateSpec, fldPath *field.Path) fi
 
 	el = append(el, validateIssuerRef(crt.IssuerRef, fldPath)...)
 
-	if len(crt.CommonName) == 0 && len(crt.DNSNames) == 0 && len(crt.URISANs) == 0 {
-		el = append(el, field.Required(fldPath.Child("commonName", "dnsNames", "uriSANs"),
-			"at least one of commonName, dnsNames, or uriSANs must be set"))
+	if len(crt.CommonName) == 0 && len(crt.DNSNames) == 0 && len(crt.URISANs) == 0 && len(crt.EmailSANs) == 0 {
+		el = append(el, field.Required(fldPath.Child("commonName", "dnsNames", "uriSANs", "emailSANs"),
```

Comment:
Hm, not sure what this `fldPath.Child` is doing.. we should probably change this to be `field.Invalid` and just pass `fldPath` itself, and include the same error description as below, e.g.:

```
field.Invalid(fldPath, "", "at least one of ...")
```

Reply:
Changed it!